### PR TITLE
merge test_conv_igemm_dynamic_small into test_conv_igemm_dynamic

### PR DESCRIPTION
**Pull request by carlushuang**
this fixes issue https://github.com/ROCmSoftwarePlatform/MIOpen/issues/648 by merge test_conv_igemm_dynamic_small into est_conv_igemm_dynamic

### CODE REVIEW

**Review by atamazov — COMMENTED** [review]
I am unable to review this as I do not understand the original issue #648. Someone please elaborate it.

**Comment by asleepzzz** [comment]
I updated log  in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/648,
this unit test is only for gfx908, we should disable this test if platform is not 908

**Comment by asroy** [comment]
> I am unable to review this as I do not understand the original issue #648. Someone please elaborate it.

Yea, this PR and original issue need more clarification
@carlushuang

**Review by atamazov — CHANGES_REQUESTED** [review]
Perhaps I do not understand something?

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -642,7 +642,7 @@ set(DYNAMIC_IMPLICITGEMM_WRW_ENVS_XDLOPS
     ${DYNAMIC_IMPLICITGEMM_COMMON}
     MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicWrwXdlops)
 
-add_custom_test(test_conv_igemm_dynamic_small
+add_custom_test(test_conv_igemm_dynamic SKIP_UNLESS_ALL
```
[the remark this answers is not in the document]
Comment (carlushuang) [reply test/CMakeLists.txt]: Hi @atamazov I think you are right. Let me just move `test_conv_igemm_dynamic` to `SKIP_TEST` when under gfx908 test.(`MIOPEN_TEST_GFX908`)

[the remark this answers is not in the document]
Comment (carlushuang) [reply test/CMakeLists.txt]: Yes, this is small test. Let me move to `SKIP_TEST` when under gfx908

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -58,10 +58,10 @@ set(MIOPEN_TEST_FLOAT_ARG)
 
 if(MIOPEN_TEST_HALF)   
     if(MIOPEN_BACKEND_OPENCL)
-        set(SKIP_TESTS test_gru test_rnn_vanilla test_lstm test_conv_igemm_dynamic)
+        set(SKIP_TESTS test_gru test_rnn_vanilla test_lstm test_conv_igemm_dynamic test_conv_igemm_dynamic_small)
     endif()
     if(MIOPEN_TEST_GFX908)
-       set(SKIP_TESTS test_immed_conv3d test_conv3d test_fusion_aux test_activation test_lrn_test test_ctc test_conv2d_bias test_conv3d_bias test_cba_inference test_cbna_inference test_pooling2d test_na_train test_na_inference test_bn_aux test_conv_igemm_dynamic)
+       set(SKIP_TESTS test_immed_conv3d test_conv3d test_fusion_aux test_activation test_lrn_test test_ctc test_conv2d_bias test_conv3d_bias test_cba_inference test_cbna_inference test_pooling2d test_na_train test_na_inference test_bn_aux test_conv_igemm_dynamic test_conv_igemm_dynamic_small)
```
Comment (atamazov) [inline test/CMakeLists.txt:64]: [Recommendation] Add `${SKIP_TESTS}` at the end to keep values added to `SKIP_TESTS` when backend is OpenCL.

**Review by atamazov — CHANGES_REQUESTED** [review]
~One mandatory comment.~ Review is irrelevant.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -58,10 +58,10 @@ set(MIOPEN_TEST_FLOAT_ARG)
 
 if(MIOPEN_TEST_HALF)   
     if(MIOPEN_BACKEND_OPENCL)
-        set(SKIP_TESTS test_gru test_rnn_vanilla test_lstm test_conv_igemm_dynamic)
+        set(SKIP_TESTS test_gru test_rnn_vanilla test_lstm test_conv_igemm_dynamic test_conv_igemm_dynamic_small)
```
[the remark this answers is not in the document]
Comment (atamazov) [reply test/CMakeLists.txt:61]: [Note] Yes, I know that we do not test OCL backend with gfx908 yet.

[the remark this answers is not in the document]
Comment (atamazov) [reply test/CMakeLists.txt:61]: Oops, the comment is wrong.

[the remark this answers is not in the document]
Comment (atamazov) [reply test/CMakeLists.txt:61]: `test_conv_igemm_dynamic` and `test_conv_igemm_dynamic_small` tests only FP32.

**Review by atamazov — CHANGES_REQUESTED** [review]
`test_conv_igemm_dynamic` and `test_conv_igemm_dynamic_small` should be only used with Vega targets and only with FP32 datatypes. This doesn't happen due to a bug in `add_custom_test`. Fix is #750 

Please confirm and close this PR.

**Comment by carlushuang** [comment]
> `test_conv_igemm_dynamic` and `test_conv_igemm_dynamic_small` should be only used with Vega targets and only with FP32 datatypes. This doesn't happen due to a bug in `add_custom_test`. Fix is #750
> 
> Please confirm and close this PR.

As per https://github.com/ROCmSoftwarePlatform/MIOpen/pull/750#discussion_r581140999, this PR should closed, but in order to remove `test_conv_igemm_dynamic` / `test_conv_igemm_dynamic_small` from gfx908 target, better refactor these custom test inside `if(NOT MIOPEN_TEST_GFX908)`
```
if(NOT MIOPEN_TEST_GFX908)
    add_custom_test(test_conv_igemm_dynamic_small
    ...
    )
endif()
```

**Comment by atamazov** [comment]
Yes, this file needs redesign.

**Pull request closed without merging**